tidb-cloud/integrate-tidbcloud-with-zapier.md

```diff
@@ -0,0 +1,259 @@
+- [Send emails via Gmail from new custom TiDB queries](https://zapier.com/apps/gmail/integrations/tidb-cloud/1134903/send-emails-via-gmail-from-new-custom-tidb-queries).
+- [Add rows to TiDB Cloud from newly caught webhooks](https://zapier.com/apps/tidb-cloud/integrations/webhook/1134955/add-rows-to-tidb-cloud-from-newly-caught-webhooks).
+- [Store new Salesforce contacts on TiDB rows](https://zapier.com/apps/salesforce/integrations/tidb-cloud/1134923/store-new-salesforce-contacts-on-tidb-rows).
+- [Create TiDB rows for new Gmail emails with resumes and send direct Slack notifications](https://zapier.com/apps/gmail/integrations/slack/1135456/create-tidb-rows-for-new-gmail-emails-with-resumes-and-send-direct-slack-notifications)
+
+## FAQ
+
+### How can I set up the TiDB Cloud account in Zapier?
+
+Zapier requires your **TiDB Cloud API key** to connect with your TiDB Cloud account. Zapier does not need your login account for TiDB Cloud.
+
+To get your TiDB Cloud API key, follow the [TiDB Cloud API documentation](https://docs.pingcap.com/tidbcloud/api/v1beta#section/Authentication/API-Key-Management).
+
+### How TiDB Cloud triggers de-duplication
+
+Zapier triggers can work with a polling API call to check for new data periodically (depends on Zapier plan).
+
+TiDB Cloud triggers provide the polling API call which will return a lot of results, most of which Zapier has seen before.
+
+Since we don’t want to trigger an action multiple times when an item in your API exists in multiple distinct polls, we will deduplicate the data with the `id` field.
+
+`New Cluster` and `New Table` can simply use the `cluster_id` and `table_id` as `id` field to do the deduplication. You need not do anything for them. Here I will introduce other triggers.
+
+**New Row Trigger**
+
+First, `New Row` trigger limits 10,000 results in every fetch. This will cause the new rows will not be triggered for they may not be included in this 10000 results.
+
+One way to avoid this is to specify `Order By` configuration in the trigger. For example, once you orderby create time, the new rows will always be included in the 10000 results.
+
+Second, `New Row` will use a flexible strategy to generate the `id` filed to do the deduplication. Here are the priority:
+
+1. `id` column if the result contains `id` column.
+2. `Dedupe Key` if you specify it in the trigger configuration.
+3. primary key if the table has a primary key (use the first column if there are multiple primary keys).
+4. unique key if the table has a unique key.
+5. the first column of the table.
+
+**New Row (Custom Query) Trigger**
+
+`New Row (Custom Query)` trigger limits 1,000,000 results in every fetch. It is a large number, and we only set it to protect the whole system. So, Your query is desired to include order and limit.
+
+As for deduplication, your query results must have a unique id field or you will get the `You must return the results with id field` error.
```

Comment:
```suggestion
To perform deduplication, your query results must have a unique id field. Otherwise, you will get the `You must return the results with id field` error.
```